src/lib/awsIamRole.js

```diff
@@ -38,25 +39,33 @@ var deleteIAMRole = function(iam, options, cb) {
 }
 */
 var createRole = function(iam, options, cb) {
-  var extid = options.externalId;
-  options.assumeRolePolicyDocument.Statement[0].Condition.StringEquals = {"sts:ExternalId": extid};
-  var params = {
+   var extid = options.externalId;
+   const awsroles = JSON.parse(fs.readFileSync('./json/federate-role_info.json', 'utf8'));
+   const createUserIAMPolicyDocument = awsroles.createUserIAMPolicyDocument;
+   options.assumeRolePolicyDocument.Statement[0].Condition.StringEquals = {"sts:ExternalId": extid};
+   var params = {
     AssumeRolePolicyDocument: JSON.stringify(options.assumeRolePolicyDocument),
     RoleName: options.roleName,
     Path: options.path
-  };
-  iam.createRole(params, function(err, data) {
-    if (err) {
-      return cb("Failed to create a role : " + err);
-    }
-    else {
-      options.roleArn = data.Role.Arn;
-      if (typeof options.policyDocument === 'undefined' || options.policyDocument === null) {
-        attachRolePolicy(iam, options, cb);
-      } else {
-        addInlineRolePolicy(iam, options, cb);
-      }
-    }
+   };
+   iam.createRole(params, function(err, data) {
+     if (err) {
+       return cb("Failed to create a role : " + err);
+     }
+     else {
+       options.roleArn = data.Role.Arn;
+       if (options.roleName == process.env.SC_ADMIN_ROLE_NAME){
```

Comment:
Kindly cook the data first and let this createRole take appropriate params to create the AWS IAM role. This function is a generic one having SC_ADMIN_ROLE_NAME check is not the correct way.

Reply:
Please review the latest changes.